### PR TITLE
Error notifications appear consistently for all create functions

### PR DESCRIPTION
**Pull request by Megan-Wright**
# Changes

For: https://github.com/tektoncd/dashboard/issues/1013

Error notifications now appear every time create functions are triggered, instead of only on the first time.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/dashboard/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

**Comment by a-roberts** [comment]
Yeah, I worked on this with Megan and determined we weren't actually passing clearNotification correctly in the case of using actions, and no rerender was occurring for the PipelineRun/PipelineResource creation pages and so by manipulating state again (doing a reset) causes a proper render again which allows our notification to pop up.

I'll test this further now to check we haven't missed any edge cases, otherwise let's get it in for 0.5.0.

File: src/containers/CreatePipelineRun/CreatePipelineRun.js [hunk 1]
```diff
@@ -105,11 +105,16 @@ const isValidLabel = (type, value) => {
   return regex.test(value);
 };
 
+function resetError() {
```
Comment (AlanGreene) [inline src/containers/CreatePipelineRun/CreatePipelineRun.js]: Is there a reason this is defined outside the component class?

Normally this would be defined like this:

```js
class CreatePipelineRun extends Component {
  constructor(props) {
    super(props);
    this.state = this.initialState(props);
  }

  resetError = () => {
    this.setState({ submitError: '' });
  }

  ...
}
```

Comment (Megan-Wright) [reply src/containers/CreatePipelineRun/CreatePipelineRun.js]: ah, nope, I've fixed that now for all of them

**Comment by a-roberts** [comment]
👍  We've got one more change for this PR to do with the success scenario too (so when we create, delete, create, we should get the second create notification), then unless anything's obviously odd I plan to take a look and get it merged for our 0.5.0 release today
